convert2rhel/actions/system_checks/convert2rhel_latest.py

```diff
@@ -161,14 +146,67 @@ def run(self):
+            (
+                "rpm",
+                "-qf",
+                running_convert2rhel_init_file,
+                "--qf",
+                "C2R %{NAME}-%|EPOCH?{%{EPOCH}}:{0}|:%{VERSION}-%{RELEASE}.%{ARCH}\n",
+            ),
+            print_output=False,
+        )
+
+        running_convert2rhel_NEVRA = _extract_convert2rhel_versions(running_convert2rhel_NEVRA)
+
+        # If any of those steps fail, we print a warning that we could not determine the rpm release and use convert2rhel.{}version{} to compare with the latest packaged version
+        if return_code != 0 or len(running_convert2rhel_NEVRA) != 1:
+            logger.warning(
+                "Couldn't determine the rpm release; We will check that the version of convert2rhel (%s) is the latest but ignore the rpm release."
+                % running_convert2rhel_version
+            )
+
+        else:
+            running_convert2rhel_NEVRA = running_convert2rhel_NEVRA[0]
+            # Run `rpm -V <convert2rhel pkg NEVRA>` to make sure the user hasn't installed a different convert2rhel version on top of a previously installed rpm package through other means than rpm (e.g. pip install from GitHub)
+            rpm_convert2rhel_verify, return_code = utils.run_subprocess(["rpm", "-V", running_convert2rhel_NEVRA])
+
+            # If any of those steps fail, we print a warning that we could not determine the rpm release and use convert2rhel.{}version{} to compare with the latest packaged version
+            if return_code != 0:
+                logger.warning(
+                    "Some files in the convert2rhel package have changed so the installed convert2rhel is not what was packaged."
+                    " We will check that the version of convert2rhel (%s) is the latest but ignore the rpm release."
+                    % running_convert2rhel_version
+                )
+
+            # Otherwise use the NEVRA from above to compare with the latest packaged version
+            else:
+                parsed_convert2rhel_version = parse_pkg_string(running_convert2rhel_NEVRA)
+
+                precise_convert2rhel_version = (
+                    parsed_convert2rhel_version[1],
+                    parsed_convert2rhel_version[2],
+                    parsed_convert2rhel_version[3],
+                )
+
+                precise_available_version = latest_available_version
+
         # After the for loop, the latest_available_version variable will gain the epoch, version, and release
         # (e.g. ("0" "0.26" "1.el7")) information from the convert2rhel yum repo
         # when the versions are the same the latest_available_version's release field will cause it to evaluate as a later version.
-        # Therefore we need to hardcode "0" for both the epoch and release below for installed_convert2rhel_version
+        # Therefore we need to hardcode "0" for both the epoch and release below for running_convert2rhel_version
```

Comment:
This whole comment block is no longer needed as it went along with where we were hardcoding zero in the rpm.labelCompare call.

Reply:
I have gotten rid of this whole comment block!